hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -520,9 +517,11 @@ private Pair<Integer, HoodieData<HoodieRecord>> initializePartitionStatsIndex(Li
   }
 
   private Pair<Integer, HoodieData<HoodieRecord>> initializeColumnStatsPartition(Map<String, Map<String, Long>> partitionToFilesMap) {
+    // during initialization, we need stats for base and log files.
     HoodieData<HoodieRecord> records = HoodieTableMetadataUtil.convertFilesToColumnStatsRecords(
         engineContext, Collections.emptyMap(), partitionToFilesMap, dataMetaClient, dataWriteConfig.isMetadataColumnStatsIndexEnabled(),
-        dataWriteConfig.getColumnStatsIndexParallelism(), dataWriteConfig.getColumnsEnabledForColumnStatsIndex());
+        dataWriteConfig.getColumnStatsIndexParallelism(), dataWriteConfig.getColumnsEnabledForColumnStatsIndex(), true,
```

Comment:
Could we have a follow-up to clean up `dataWriteConfig.getColumnsEnabledForColumnStatsIndex()`?  This method is only called when col stats is enabled.  It's redundant to pass this in again.

Reply:
not sure whats the follow up here. can you throw some light please

Reply:
This method `convertFilesToColumnStatsRecords` is called when column stats index is enabled and the MDT records need to be generated.  It is useless to check `dataWriteConfig.getColumnsEnabledForColumnStatsIndex()` again.

Reply:
not sure I get your feedback. dataWriteConfig.getColumnsEnabledForColumnStatsIndex() is used to fetch the list of columns to index in col stats partition in MDT. not sure how it can be useless. can you help me understand.

Reply:
I was referring to dataWriteConfig.isMetadataColumnStatsIndexEnabled(). This argument looks redundant.